elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java

```diff
@@ -156,14 +151,32 @@ private void populateMetaData(MetaDataStore metaDataStore) {
                 .forEach(metaDataStore::addTable);
     }
 
+    public interface Transaction extends AutoCloseable {
+        @Override
+        void close();
```

Comment:
Same.  Also we are adding a cancel method in another PR.  Do see these as the same thing?  For cancel, the expectation is the method is thread safe.  If we consolidate them - we'll need the same comment here.

Reply:
I assume if cancel() is called on a AggregationDataStoreTransaction that it ends the whole transaction, not just any pending calls such as loadObjects? (There won't be a use case for aborting loadObjects but still being able to use the same AggregationDataStoreTransaction after.)